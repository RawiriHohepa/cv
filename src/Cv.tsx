import React from "react";
import styled from "styled-components";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import Entry from "./components/Entry";
import {
    Title,
    SubTitle,
    SectionTitle,
    Text,
    Bold,
} from "./components/text";
import SectionWithDetails from "./components/SectionWithDetails";
import SectionWithText from "./components/SectionWithText";
import {
    contacts,
    experience,
    projects,
    skills,
    leadership,
} from "./data";

const Paper = styled.div`
    // A4 size paper for 96 PPI
    width: 794px;
    height: 1123px;
`

const HeaderSection = styled.div`
    height: 10%;
    width: 100%;
    box-sizing: border-box;
    padding: 2rem;

    background: #59BCDB80;

    display: flex;
    justify-content: space-between;
`
const ContactsSection = styled.div`
    display: flex;
    flex-direction: column;
`
const Content = styled.div`
    height: 90%;
    display: flex;
`
const LeftSection = styled.div`
    width: 60%;
    height: 100%;
    box-sizing: border-box;
    padding: 2rem;
    & > :first-child {
        margin-top: 0;
    }
`
const RightSection = styled.div`
    width: 40%;
    height: 100%;
    box-sizing: border-box;
    padding: 1.5rem;
    & > :first-child {
        margin-top: 0;
    }

    background: #B3E3F180;
`

const ContactsItem = styled.a`
    color: #000000;
    text-align: right;
    margin-left: 0.5rem;
`

const Cv = () => {
    return (
        <Paper>
            <HeaderSection>
                <div>
                    <Title>Rawiri Hohepa</Title>
                    <SubTitle>Final Year Software Engineering Student</SubTitle>
                </div>
                <ContactsSection>
                    {contacts.map(contact => (
                        <div style={{ display: "flex", justifyContent: "space-between", alignItems: "center", marginBottom: "0.25rem" }} key={contact.name}>
                            <FontAwesomeIcon icon={contact.icon} />
                            <ContactsItem href={contact.link}>{contact.name}</ContactsItem>
                        </div>
                    ))}
                </ContactsSection>
            </HeaderSection>

            <Content>
                <LeftSection>
                    <SectionWithDetails
                        title="Experience"
                        entries={experience}
                    />

                    <SectionWithDetails
                        title="Projects"
                        entries={projects}
                    />

                    <SectionWithText
                        title="Ask me about..."
                        entries={skills}
                    />
                </LeftSection>


                <RightSection>
                    <SectionTitle>Education</SectionTitle>

                    <Entry
                        title="University of Auckland | BE (Hons) & BA"
                        subtitle="Software Engineering, Psychology, Māori Studies"
                        description="Mar 2018 — Nov 2022"
                    >
                        <Text>GPA 8.6/9 (<Bold>A to A+</Bold> average)</Text>
                        <Text>Awards:</Text>
                        <div style={{ marginLeft: "1.5rem" }}>
                            <Text style={{ color: "#4B4B4B" }}>Dean’s Honours List | 2018, 2019, 2021</Text>
                            <Text style={{ color: "#4B4B4B" }}>Māori Academic Excellence Scholarship | 2017</Text>
                        </div>
                    </Entry>

                    <SectionWithDetails
                        title="Leadership & Community"
                        entries={leadership}
                        size="small"
                    />

                    <SectionTitle>References</SectionTitle>

                    <Text>Available upon request</Text>
                </RightSection>
            </Content>
        </Paper>
    );
}

export default Cv;
