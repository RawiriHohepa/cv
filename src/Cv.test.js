import React from 'react';
import renderer from 'react-test-renderer';
import Cv from './Cv';

test('should render correctly', () => {
  const component = renderer.create(
    <Cv />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
