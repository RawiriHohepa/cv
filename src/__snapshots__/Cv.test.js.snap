// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="sc-bdvvtL fEOOyA"
>
  <div
    className="sc-gsDKAQ fYFnUa"
  >
    <div>
      <h1>
        Rawiri Hohepa
      </h1>
      <h1>
        Final Year Software Engineering Student
      </h1>
    </div>
    <div
      className="sc-dkPtRN fKsULa"
    >
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "space-between",
            "marginBottom": "0.25rem",
          }
        }
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-github "
          data-icon="github"
          data-prefix="fab"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 496 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3 .3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5 .3-6.2 2.3zm44.2-1.7c-2.9 .7-4.9 2.6-4.6 4.9 .3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3 .7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3 .3 2.9 2.3 3.9 1.6 1 3.6 .7 4.3-.7 .7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3 .7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3 .7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <a
          className="sc-gKclnd kLtOtz"
          href="https://github.com/rawirihohepa"
        >
          RawiriHohepa
        </a>
      </div>
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "space-between",
            "marginBottom": "0.25rem",
          }
        }
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-envelope "
          data-icon="envelope"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 352c-16.53 0-33.06-5.422-47.16-16.41L0 173.2V400C0 426.5 21.49 448 48 448h416c26.51 0 48-21.49 48-48V173.2l-208.8 162.5C289.1 346.6 272.5 352 256 352zM16.29 145.3l212.2 165.1c16.19 12.6 38.87 12.6 55.06 0l212.2-165.1C505.1 137.3 512 125 512 112C512 85.49 490.5 64 464 64h-416C21.49 64 0 85.49 0 112C0 125 6.01 137.3 16.29 145.3z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <a
          className="sc-gKclnd kLtOtz"
          href="mailto:rawirihohepa@gmail.com"
        >
          rawirihohepa@gmail.com
        </a>
      </div>
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "space-between",
            "marginBottom": "0.25rem",
          }
        }
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-phone "
          data-icon="phone"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M511.2 387l-23.25 100.8c-3.266 14.25-15.79 24.22-30.46 24.22C205.2 512 0 306.8 0 54.5c0-14.66 9.969-27.2 24.22-30.45l100.8-23.25C139.7-2.602 154.7 5.018 160.8 18.92l46.52 108.5c5.438 12.78 1.77 27.67-8.98 36.45L144.5 207.1c33.98 69.22 90.26 125.5 159.5 159.5l44.08-53.8c8.688-10.78 23.69-14.51 36.47-8.975l108.5 46.51C506.1 357.2 514.6 372.4 511.2 387z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <a
          className="sc-gKclnd kLtOtz"
          href="tel:028 407 9883"
        >
          028 407 9883
        </a>
      </div>
      <div
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "justifyContent": "space-between",
            "marginBottom": "0.25rem",
          }
        }
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-linkedin "
          data-icon="linkedin"
          data-prefix="fab"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <a
          className="sc-gKclnd kLtOtz"
          href="https://www.linkedin.com/in/rawiri-hohepa/"
        >
          rawiri-hohepa
        </a>
      </div>
    </div>
  </div>
  <div
    className="sc-hKwDye iEBwvH"
  >
    <div
      className="sc-eCImPb hchGxK"
    >
      <h2
        className="undefined undefined"
      >
        Experience
      </h2>
      

      <div
        style={
          Object {
            "marginBottom": "1rem",
          }
        }
      >
        <h3
          className="undefined undefined"
        >
          Full Stack Developer | Maverick
        </h3>
        <h4
          className="undefined undefined"
        >
          Nov 2021 — Feb 2022 (4 months)
        </h4>
        <ul>
          

          <li
            className="undefined undefined"
          >
            Created 
            <strong>
              React Typescript
            </strong>
             webapp from scratch for a digital pilates platform, and integrated with 
            <strong>
              C#
            </strong>
             API
          </li>
          

          <li
            className="undefined undefined"
          >
            Created 
            <strong>
              NextJS Typescript
            </strong>
             webapp MVP from scratch for a non-profit food organisation, and integrated with Strapi 
            <strong>
              Content Management System (CMS)
            </strong>
          </li>
          

        </ul>
      </div>
      

      <div
        style={
          Object {
            "marginBottom": "1rem",
          }
        }
      >
        <h3
          className="undefined undefined"
        >
          Software Developer Intern | Pocketful
        </h3>
        <h4
          className="undefined undefined"
        >
          Aug 2020 — Feb 2021 (7 months)
        </h4>
        <ul>
          

          <li
            className="undefined undefined"
          >
            Overhauled 
            <strong>
              React Native
            </strong>
             user interface using newly created components written in 
            <strong>
              Typescript
            </strong>
          </li>
          

          <li
            className="undefined undefined"
          >
            Liaised with and integrated external company in 
            <strong>
              C# & React Native
            </strong>
             proof-of-concept to reward users with digital vouchers
          </li>
          

        </ul>
      </div>
      

      <div
        style={
          Object {
            "marginBottom": "1rem",
          }
        }
      >
        <h3
          className="undefined undefined"
        >
          Software Engineer Intern | ASB Bank
        </h3>
        <h4
          className="undefined undefined"
        >
          Nov 2019 — Feb 2020 (4 months)
        </h4>
        <ul>
          

          <li
            className="undefined undefined"
          >
            Designed and created 
            <strong>
              React
            </strong>
             webapp and 
            <strong>
              C#
            </strong>
             Web API to store, retrieve, and display international payment updates
          </li>
          

        </ul>
      </div>
      <h2
        className="undefined undefined"
      >
        Projects
      </h2>
      

      <div
        style={
          Object {
            "marginBottom": "1rem",
          }
        }
      >
        <h3
          className="undefined undefined"
        >
          Online Cards Against Humanity
        </h3>
        <h4
          className="undefined undefined"
        >
          <a
            href="https://nerdsopposesociety.xyz/"
            style={
              Object {
                "color": "#000000",
              }
            }
          >
            nerdsopposesociety.xyz
          </a>
           | GitHub: 
          <a
            href="https://github.com/SE750-DART/nerds-oppose-society"
            style={
              Object {
                "color": "#000000",
              }
            }
          >
            nerds-oppose-society
          </a>
        </h4>
        <h4
          className="undefined undefined"
        >
          May 2021
        </h4>
        <ul>
          

          <li
            className="undefined undefined"
          >
            Created a responsive, mobile-first webapp of Cards Against Humanity for a university project, earning a 
            <strong>
              100% mark
            </strong>
          </li>
          

          <li
            className="undefined undefined"
          >
            Integrated 
            <strong>
              React
            </strong>
             frontend system with 
            <strong>
              SocketJS
            </strong>
             & 
            <strong>
              ExpressJS
            </strong>
             backend systems written in 
            <strong>
              Typescript
            </strong>
          </li>
          

        </ul>
      </div>
      

      <div
        style={
          Object {
            "marginBottom": "1rem",
          }
        }
      >
        <h3
          className="undefined undefined"
        >
          Responsible Bar Searcher
        </h3>
        <h4
          className="undefined undefined"
        >
          GitHub: 
          <a
            href="https://github.com/Alxjlv/Sloshed"
            style={
              Object {
                "color": "#000000",
              }
            }
          >
            Sloshed
          </a>
        </h4>
        <h4
          className="undefined undefined"
        >
          Aug 2020
        </h4>
        <ul>
          

          <li
            className="undefined undefined"
          >
            Led backend development on HTML/CSS & Javascript webapp to display nearby bars and liquor stores using 
            <strong>
              Google Places & Geolocation APIs
            </strong>
          </li>
          

          <li
            className="undefined undefined"
          >
            <strong>
              Won Best Design
            </strong>
             at Summer of Tech's 2020 Create Camp Hackathon
          </li>
          

        </ul>
      </div>
      

      <div
        style={
          Object {
            "marginBottom": "1rem",
          }
        }
      >
        <h3
          className="undefined undefined"
        >
          Investments Tracker
        </h3>
        <h4
          className="undefined undefined"
        >
          GitHub: 
          <a
            href="https://github.com/RawiriHohepa/Investments-Tracker"
            style={
              Object {
                "color": "#000000",
              }
            }
          >
            Investments-Tracker
          </a>
        </h4>
        <h4
          className="undefined undefined"
        >
          Ongoing
        </h4>
        <ul>
          

          <li
            className="undefined undefined"
          >
            Created 
            <strong>
              ExpressJS
            </strong>
             API written in 
            <strong>
              Typescript
            </strong>
             to automatically retrieve and collate personal investments using 
            <strong>
              web scraping
            </strong>
             & external APIs
          </li>
          

          <li
            className="undefined undefined"
          >
            Future plans to automatically call API periodically, store results in a 
            <strong>
              MongoDB
            </strong>
             database, and display helpful statistics in a 
            <strong>
              React
            </strong>
             webapp
          </li>
          

        </ul>
      </div>
    </div>
    <div
      className="sc-jRQBWg kvZXnF"
    >
      <h2
        className="undefined undefined"
      >
        Education
      </h2>
      <div
        style={
          Object {
            "marginBottom": "1rem",
          }
        }
      >
        <h3
          className="undefined undefined"
        >
          University of Auckland | BE (Hons) & BA
        </h3>
        <h4
          className="undefined undefined"
        >
          Software Engineering, Psychology, Māori Studies
        </h4>
        <h4
          className="undefined undefined"
        >
          Mar 2018 — Nov 2022
        </h4>
        <p
          className="undefined undefined"
        >
          GPA 8.6/9 (
          <span>
            A to A+
          </span>
           average)
        </p>
        <p
          className="undefined undefined"
        >
          Awards:
        </p>
        <div
          style={
            Object {
              "marginLeft": "1.5rem",
            }
          }
        >
          <p
            className="undefined undefined"
            style={
              Object {
                "color": "#4B4B4B",
              }
            }
          >
            Dean’s Honours List | 2018, 2019, 2021
          </p>
          <p
            className="undefined undefined"
            style={
              Object {
                "color": "#4B4B4B",
              }
            }
          >
            Māori Academic Excellence Scholarship | 2017
          </p>
        </div>
      </div>
      <h2
        className="undefined undefined"
      >
        Leadership & Community
      </h2>
      

      <div
        style={
          Object {
            "marginBottom": "0.5rem",
          }
        }
      >
        <h3
          className="undefined undefined"
        >
          Secretary
        </h3>
        <h4
          className="undefined undefined"
        >
          Software Engineering Students Association (SESA) | 2022
        </h4>
        <ul>
          

          <li
            className="undefined undefined"
          >
            Overhaul meeting and task tracking system to comply with changing university policy
          </li>
          

          <li
            className="undefined undefined"
          >
            Uphold and enforce SESA constitution
          </li>
          

          <li
            className="undefined undefined"
          >
            Organise and record executive commitee meetings
          </li>
          

        </ul>
      </div>
      

      <div
        style={
          Object {
            "marginBottom": "0.5rem",
          }
        }
      >
        <h3
          className="undefined undefined"
        >
          Part IV Representative
        </h3>
        <h4
          className="undefined undefined"
        >
          Software Engineering Students Association (SESA) | 2021
        </h4>
        <ul>
          

          <li
            className="undefined undefined"
          >
            Organised and MC’d SESA’s inaugural hackathon, with various industry professionals and &gt;100 students
          </li>
          

          <li
            className="undefined undefined"
          >
            Represented SESA at student wellbeing panel for all engineering student associations
          </li>
          

          <li
            className="undefined undefined"
          >
            Organised and ran various industry & social events for software engineering students
          </li>
          

        </ul>
      </div>
      

      <div
        style={
          Object {
            "marginBottom": "0.5rem",
          }
        }
      >
        <h3
          className="undefined undefined"
        >
          Frontend Team Lead
        </h3>
        <h4
          className="undefined undefined"
        >
          20-Person University Project | Apr 2021
        </h4>
        <ul>
          

          <li
            className="undefined undefined"
          >
            Managed frontend team in project to create a digital bullet journal webapp, earning an A grade
          </li>
          

          <li
            className="undefined undefined"
          >
            Organised and led team meetings, liased with design team, assigned tasks to team members
          </li>
          

        </ul>
      </div>
      

      <div
        style={
          Object {
            "marginBottom": "0.5rem",
          }
        }
      >
        <h3
          className="undefined undefined"
        >
          Peer Mentor
        </h3>
        <h4
          className="undefined undefined"
        >
          Software Engineering Students Association (SESA) | 2021
        </h4>
        <ul>
          

          <li
            className="undefined undefined"
          >
            Mentored a Part II (2nd year) software engineering student through SESA’s mentoring program
          </li>
          

          <li
            className="undefined undefined"
          >
            Provided advice & guidance regarding university courses, internships, recruitment, study/life balance, etc.
          </li>
          

        </ul>
      </div>
      

      <div
        style={
          Object {
            "marginBottom": "0.5rem",
          }
        }
      >
        <h3
          className="undefined undefined"
        >
          Tutor
        </h3>
        <h4
          className="undefined undefined"
        >
          UoA Part I & II Assistance Centres | 2019, 2020
        </h4>
        <ul>
          

          <li
            className="undefined undefined"
          >
            Supported & tutored Part I & II (1st & 2nd year) engineering students
          </li>
          

          <li
            className="undefined undefined"
          >
            Facilitated understanding of lecture notes and preparation for tests, assignments, and exams
          </li>
          

        </ul>
      </div>
      <h2
        className="undefined undefined"
      >
        Ask me about...
      </h2>
      

      <p
        className="undefined undefined"
      >
        <strong>
          Frontend:
        </strong>
         React, React Native, NextJS, TypeScript, JavaScript, HTML/CSS
      </p>
      

      <p
        className="undefined undefined"
      >
        <strong>
          Backend:
        </strong>
         ExpressJS, NodeJS, MongoDB, C# Entity Framework, Java
      </p>
      

      <p
        className="undefined undefined"
      >
        <strong>
          Design:
        </strong>
         Figma
      </p>
      

      <p
        className="undefined undefined"
      >
        <strong>
          Version control:
        </strong>
         Git, GitHub
      </p>
      

      <p
        className="undefined undefined"
      >
        <strong>
          Languages:
        </strong>
         TypeScript, JavaScript, Java, C#, SQL
      </p>
      

      <p
        className="undefined undefined"
      >
        <strong>
          Interests:
        </strong>
         Tabletop roleplaying, Video games, Finances
      </p>
      <h2
        className="undefined undefined"
      >
        References
      </h2>
      

      <p
        className="undefined undefined"
      >
        Available upon request
      </p>
    </div>
  </div>
</div>
`;
